packages/block-library/src/navigation-menu/index.php

```diff
@@ -15,20 +15,82 @@
  * @return string Returns the post content with the legacy widget added.
  */
 function render_block_navigation_menu( $attributes, $content, $block ) {
-	return '<nav class="wp-block-navigation-menu">' . build_navigation_menu_html( $block ) . '</nav>';
+	// CSS classes.
+	$colors = array(
```

Comment:
@retrofox could we move all this CSS computation into its own function please? That way it would be like the HTML rendering at the end.

Reply:
:+1:

Reply:
handled here https://github.com/WordPress/gutenberg/pull/17832/commits/ddc4311bfecbb2206f3a14c7e538be48739810c8